### DISCUSSION TITLE
wmbusmeters not receiving anything

### DISCUSSION THREAD

**Discussion by Smeden-dk** [wmbusmeters/wmbusmeters#1161] [Q&A] [answered]
I am trying to read wmbus telegrams from my Kamstrup Multical 602 heat meter. The log file from HA says:

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun nginx (no readiness notification)
s6-rc: info: service legacy-services successfully started
[16:44:09] INFO: CONFIG_CONF ...
[16:44:09] INFO: {"device":"/dev/ttyUSB0:im871a:c1,t1","donotprobe":"/dev/ttyAMA0","format":"json","logfile":"/dev/stdout","loglevel":"normal","logtelegrams":false,"shell":"/wmbusmeters/mosquitto_pub.sh \"wmbusmeters/$METER_NAME\" \"$METER_JSON\""}
[16:44:09] INFO: CONFIG_METERS ...
[16:44:09] INFO: [{"driver":"","id":"","key":"","name":""}]
[16:44:09] INFO: Syncing wmbusmeters configuration ...
[16:44:09] INFO: Registering meters ...
[16:44:09] INFO: Adding meter-0001 ...
[16:44:09] INFO: Generating MQTT configuration ... 
[16:44:09] INFO: Broker core-mosquitto will be used.
[16:44:09] INFO: 
MQTT Discovery ...
tar: wmbusmeters-master/ha-addon/mqtt_discovery: not found in archive
[16:44:10] INFO:  Adding meter:  ...
[16:44:10] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/.json not found.
[16:44:10] INFO: MQTT Discovery cleanup...
[16:44:11] INFO: Starting web configuration service.
[16:44:11] INFO: Running wmbusmeters ...
Not a valid meter id nor a valid meter match expression ""
Started config im871a[00104452] on /dev/ttyUSB0 listening on t1,c1
No meters configured. Printing id:s of all telegrams heard!
[16:44:12] INFO: Starting Nginx...
2024/01/28 16:44:12 [notice] 83#83: using the "epoll" event method
2024/01/28 16:44:12 [notice] 83#83: nginx/1.24.0
2024/01/28 16:44:12 [notice] 83#83: OS: Linux 5.10.0-21-amd64
2024/01/28 16:44:12 [notice] 83#83: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2024/01/28 16:44:12 [notice] 83#83: start worker processes
2024/01/28 16:44:12 [notice] 83#83: start worker process 239

Any ideas?

**Comment by 1beamer** [comment]
Hi,

I am not an expert at all but as I'm busy with my multical303 these days maybe I can help
First you have to go un WEB UI when you open the addon and fill all the necessary fields

[16:44:09] INFO: [{"driver":"","id":"","key":"","name":""}]      driver, id, key and name- you can read on documentation about all of them.

You have to specify the driver on WEB UI, in your case is "kamheat" (or you can try also with multical21). The driver file has to be placed on /config/wmbusmeters/etc/mqtt_discovery/ as the addon is using it.

Kamheat.json file can be downloaded from here:
https://github.com/wmbusmeters/wmbusmeters-ha-addon/blob/main/wmbusmeters-ha-addon-edge/mqtt_discovery/kamheat.json

**Reply by Smeden-dk** [reply]
Do you mean HA when you say Web UI? I am running wmbusmeters directly from linux to test it. I guess it should receive raw telegrams in any case?

**Reply by 1beamer** [reply]
![image](https://github.com/wmbusmeters/wmbusmeters/assets/156225428/b1e3eb70-6175-4374-a13b-8bb0213c0693)

I was referring to the addon main page from home assistant interface. I do not know how it looks on linux (see attached picture on home assistant)
As far as I have seen the 602 is not by default wmbus, it must have an adapter card, worth checking.
Also in my case the addon did not work without those parameters, did not started until I had them all figured. Key was a tricky one especially.

**Comment by BIBOLV** [comment, marked as answer]
Based on first log you provided - it is from ha-addon, so it is bit confusing if you are running standalone version. Secondly you should examine your meter if it is capable and sending wmbus data, from [docs](https://documentation.kamstrup.com/docs/MULTICAL_602/en-GB/Technical_description/CONT90C6D93022A54253A84D8345A5325B55/?page=12) you should check 6-7 position of meters type and it should be between 30-39. If it is then you should be able to capture something if meter is in range. Do not specify any meter in config to display in logs all transmissions that your dongle can hear.

**Comment by Smeden-dk** [comment]
Thank you so much 1beamer and BIBOLV. Looking in that docs, it turns out my meter is not wmbus, it has code 84 for "High power RadioRouter + pulse inputs".
